src/application/tester.go

```diff
@@ -53,21 +53,23 @@ func (a *Tester) RunAppTests() error {
 		}
 	}
 	if numFailed == 0 {
-		return a.Logger.Log("exo-test", "All tests passed", true)
+		return true, a.Logger.Log("exo-test", "All tests passed", true)
 	}
-	return a.Logger.Log("exo-test", fmt.Sprintf("%d tests failed", numFailed), true)
+	return false, a.Logger.Log("exo-test", fmt.Sprintf("%d tests failed", numFailed), true)
 }
 
 // RunServiceTest runs the tests for a single service
-func (a *Tester) RunServiceTest(serviceName string) error {
+func (a *Tester) RunServiceTest(serviceName string) (bool, error) {
+	testsPassed := true
 	if a.InternalServiceConfigs[serviceName].Tests == "" {
 		a.logChannel <- fmt.Sprintf("%s has no tests, skipping", serviceName)
 	} else {
 		if _, err := a.runServiceTests(serviceName, a.InternalServiceConfigs[serviceName]); err != nil {
```

Comment:
`runServiceTests` returns whether or not service tests pass. I believe we need to use that value. If you add a test case for running the tests for a single service where the tests fail, I believe you would currently get an exit code of 0 when we want 1.